search/includes/classes/class-search.php

```diff
@@ -18,6 +18,51 @@ class Search {
 
 	private const MAX_SEARCH_LENGTH = 255;
 
+	private const DISABLE_POST_META_ALLOW_LIST = array(
+		2341,
+	);
+
+	// From https://github.com/Automattic/jetpack/blob/c36432aa890dc24cafee4c4362711ffcafb9c983/packages/sync/src/class-defaults.php#L689-L732
+	public const POST_META_DEFAULT_ALLOW_LIST = array(
```

Comment:
I'm not sure we should copy-pasta from Jetpack like that, some of it is not necessarily useful for our use cases.

Reply:
We needed a default and decided to just use what Jetpack was using so the behaviour matches up a bit better for our migration path.

Reply:
I don't think it's the correct approach, I struggle to imagine a client searching for the data pertaining to a Restaraunt module or `_jetpack_post_thumbnail` which is only relevant to WP.com.

Once the list is set in stone it's unlikely to change, and that's how we get cruft that no one wants to touch and nobody is sure why it's there. 

My vote is to only allow Core WP meta and not bother with Jetpack-specific keys.

Reply:
We went with a default of empty and then we'll flesh it out as people use it and when we start thinking about migration path.